audio/filter/af_lavcac3enc.c

```diff
@@ -342,13 +354,33 @@ static struct mp_filter *af_lavcac3enc_create(struct mp_filter *parent,
             mp_autoconvert_add_afmt(conv, mpfmt);
     }
 
+#if !HAVE_AV_CHANNEL_LAYOUT
     const uint64_t *lch = s->lavc_acodec->channel_layouts;
-    for (int n = 0; lch && lch[n]; n++) {
+#define SET_LAYOUT(lch) (lch[n])
+#else
+    const AVChannelLayout *lch = s->lavc_acodec->ch_layouts;
+#define SET_LAYOUT(lch) (lch[n].nb_channels)
+#endif
+    for (int n = 0; lch && SET_LAYOUT(lch); n++) {
         struct mp_chmap chmap = {0};
+
+#if !HAVE_AV_CHANNEL_LAYOUT
         mp_chmap_from_lavc(&chmap, lch[n]);
+#else
+        if (!mp_chmap_from_av_layout(&chmap, &lch[n])) {
+            char layout[128] = {0};
+            MP_VERBOSE(f, "Skipping unsupported channel layout: %s\n",
+                       av_channel_layout_describe(&lch[n],
+                                                  layout, 128) < 0 ?
+                       "undefined" : layout);
+            continue;
+        }
+#endif
```

Comment:
wm4 would probably say this is too much ifdeffery
can this be reasonably extracted into a function?

Reply:
The original idea of this was to keep the same structure and just switch definition of `lch` as well as the exact check done under the macro.

The largest difference comes from the fact that at least API-wise AVChannelLayout can now support layouts which cannot be successfully converted into `mp_chmap`, so I added logging & skipping of that (`continue`).

Since the following lines are the same (`mp_autoconvert_add_chmap` etc), and you anyways have to continue in this loop, I'm not really sure what'd be the best way to split this into its own function.

Reply:
Well, one way would be to split this into two definitions of a function that adds those autoconvert chmaps, one of which is the original loop. Both take AVCodec in, and then can access the bit from it that's needed.

And then the calling side would be the same, just one or the other version of the function is defined.